docs/modules/operation/pages/provisioning/hardware-inventory.adoc

```diff
@@ -0,0 +1,111 @@
+
+IMPORTANT: To use this feature you have to install the SNMP hardware inventory plugin from the repository.
+
+****
+To install the opennms-plugin-provisioning-snmp-hardware-inventory package:
+
+* For RPM-based distributions:
+** `yum install opennms-plugin-provisioning-snmp-hardware-inventory`
+* For Debian-based distributions:
+** `apt-get install opennms-plugin-provisioning-snmp-hardware-inventory`
+****
+
+The package creates the configuration file `$OPENNMS_HOME/etc/snmp-hardware-inventory-adapter-configuration.xml` which defines the support for the https://tools.cisco.com/Support/SNMP/do/BrowseMIB.do?mibName=ENTITY-MIB[CISCO-ENTITY-MIB] as a starter example.
+
+Any MIB column indexed by `entPhysicalIndex` can be added to the configuration file and the hardware inventory adapter will collect the value and store it for each entity that contains a non-null value for the custom attribute.
+The `hw-extension` element lets you group a list of Entity-MIB extensions defined in `mibObj` elements based on a sysOidMask of a node.
+If the system enterprise ID of the node matches the `sysOidMask` defined on the `hw-extension`, the `mibObj` entities inside are collected for populating the inventory.
+When a vendor provides the standard objects on a different table rather than `entPhysicalTable`, you can override one of the base attributes by specifying on the `mibObj` that the attribute that must be overridden.
+
+
+[source, xml]
+----
+<hw-extension name="CISCO-ENTITY-ASSET-MIB" sysOidMask=".1.3.6.1.4.1.9.">
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.1"  type="string" alias="ceAssetOEMString" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.2"  type="string" alias="ceAssetSerialNumber" replace="entPhysicalSerialNum" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.3"  type="string" alias="ceAssetOrderablePartNumber" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.4"  type="string" alias="ceAssetHardwareRevision" replace="entPhysicalHardwareRev" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.5"  type="string" alias="ceAssetMfgAssyNumber" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.6"  type="string" alias="ceAssetMfgAssyRevision" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.7"  type="string" alias="ceAssetFirmwareID" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.8"  type="string" alias="ceAssetFirmwareRevision" replace="entPhysicalFirmwareRev" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.9"  type="string" alias="ceAssetSoftwareID" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.10" type="string" alias="ceAssetSoftwareRevision" replace="entPhysicalSoftwareRev" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.11" type="string" alias="ceAssetCLEI" />
+    <mibObj oid=".1.3.6.1.4.1.9.9.92.1.1.1.12" type="string" alias="ceAssetAlias" />
+</hw-extension>
+----
+
+In this example, only the nodes that match the `sysOidMask .1.3.6.1.4.1.9` (Cisco devices) are candidates to discover the `mibObj` entities defined on the extension group.
```

Comment:
Hardware discovery will always happen using the content of `ENTITY-MIB::entPhysicalTable` when the adapter is installed. What we see in the configuration file is optional and only required when you want to "extend" what is already provided via `entPhysicalTable`.

The above gives me the impression that you must add the `hw-extension` to get the adapter working, and that's not the case. Even Cisco devices would have data without that configuration, as long as they support the ENTITY-MIB. With that configuration, you'd be "enhancing" the existing content with the vendor-specific tables indexed by the `entPhysicalIndex`.

Makes sense?

Reply:
I just pushed an update to line 28 to clarify this fact.